src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -166,20 +166,30 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
             }
             catch (Exception ex)
             {
-                Logger.ErrorProcessingMessage(ex);
-
-                var authenticationFailedContext = new AuthenticationFailedContext(Context, Options)
+                try
                 {
-                    Exception = ex
-                };
+                    Logger.ErrorProcessingMessage(ex);
+
+                    var authenticationFailedContext = new AuthenticationFailedContext(Context, Options)
+                    {
+                        Exception = ex
+                    };
 
-                await Options.Events.AuthenticationFailed(authenticationFailedContext);
-                if (authenticationFailedContext.CheckEventResult(out result))
+                    await Options.Events.AuthenticationFailed(authenticationFailedContext);
+                    if (authenticationFailedContext.CheckEventResult(out result))
+                    {
+                        return result;
+                    }
+                    else
+                    {
+                        return AuthenticateResult.Fail(ex);
+                    }
+                }
+                catch (Exception ex2)
```

Comment:
Why are we doing this? This is strictly user code, they can handle their own exceptions.